spec/emittr_spec.rb

```diff
@@ -103,6 +103,93 @@
     end
   end
 
+  describe '#on_any' do
+    let(:block)     { Proc.new {} }
+    let(:any_block) { Proc.new {} }
+
+    context 'when #emit is called' do
+      it 'calls #on_any blocks' do
+        emitter.on_any &any_block
+
+        emitter.on :first_any_test, &block
+        emitter.on :second_any_test, &block
+
+        expect(any_block).to receive(:call).with(:first_any_test).exactly(2).times
+        expect(any_block).to receive(:call).with(:second_any_test).exactly(2).times
+
+        2.times { emitter.emit :first_any_test }
+        2.times { emitter.emit :second_any_test }
+      end
+    end
+  end
+
+  describe '#off_any' do
+    let(:block) { Proc.new {} }
+
+    it 'removes listener from #any list' do
+      emitter.on(:off_any_test) { Proc.new {} }
+      listeners = emitter.send(:listeners)
+
+      emitter.on_any &block
+      emitter.emit :off_any_test
+      expect(listeners[:*]).not_to be_empty
+
+      emitter.off_any &block
+      emitter.emit :off_any_test
+      expect(listeners[:*]).to be_empty
+    end
+
+    it "doesn't call block" do
+      emitter.on(:off_any_test) { Proc.new {} }
+
+      emitter.on_any &block
+      expect(block).to receive(:call)
+      emitter.emit :off_any_test
+
+      emitter.off_any &block
+      expect(block).not_to receive(:call)
+      emitter.emit :off_any_test
+    end
+  end
+
+  describe '#once_any' do
+    let(:block)    { Proc.new {} }
+    let(:first_on_block) { Proc.new {} }
+    let(:second_on_block) { Proc.new {} }
+
+    it 'adds listener to #any listeners list' do
+      expect {
+        emitter.once_any &block
+      }.to change(emitter.send(:listeners)[:*], :count).by 1
+    end
+
+    context 'when any event is emitted' do
+
+      it 'calls block added to any list only once' do
+        emitter.once_any &block
+
+        emitter.on :first_any_test, &first_on_block
```

Comment:
just to make sure that the block is being called once, but it isn't that necessary.